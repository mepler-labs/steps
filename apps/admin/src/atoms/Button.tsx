import React from 'react';
import styled from 'styled-components';

import {
  darken,
  green,
  white,
} from 'styles/colors';

interface Props {
  className?: string;
  children: any;
  white?: boolean;
}

class Button extends React.Component<Props, {}> {
  static defaultProps = { white: false };

  render() {
    const { className, children, white } = this.props;

    const ButtonEl = white ? WhiteButton : GreenButton;

    return <ButtonEl>{children}</ButtonEl>;
  }
}

const BaseButton = styled.button`
  border: none;
  border-radius: 1000px;
  cursor: pointer;
  display: inline-block;
  font-size: 0.825em;
  min-width: 180px;
  padding-bottom: 1.5em;
  padding-left: 2.25em;
  padding-right: 2.25em;
  padding-top: 1.5em;
  text-transform: uppercase;
`;

const WhiteButton = BaseButton.extend`
  background-color: ${white};
  color: ${green};
`;

const GreenButton = BaseButton.extend`
  background-color: ${green};
  color: ${white};

  &:hover {
    background-color: ${darken(green, 0.25)};
  }
`;

export default Button;
