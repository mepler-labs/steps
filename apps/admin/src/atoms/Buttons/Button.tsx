import React from 'react';
import styled from 'styled-components';

import { remCalc } from 'styles/type';
import { darken, green, white } from 'styles/colors';

interface Props {
  white?: boolean;
  type?: string;
  onClick?(event: any): void;
}

const Button: React.SFC<Props> = ({ white = false, children }) => {
  const ButtonEl = white ? WhiteButton : GreenButton;

  return <ButtonEl>{children}</ButtonEl>;
};

const BaseButton = styled.button`
  border: none;
  border-radius: 1000px;
  cursor: pointer;
  display: inline-block;
  font-size: 0.825em;
  min-width: 180px;
  padding-bottom: ${remCalc(20)};
  padding-left: ${remCalc(30)};
  padding-right: ${remCalc(30)};
  padding-top: ${remCalc(20)};
  text-transform: uppercase;
`;

const WhiteButton = BaseButton.extend`
  background-color: ${white};
  color: ${green};
`;

const GreenButton = BaseButton.extend`
  background-color: ${green};
  color: ${white};

  &:hover {
    background-color: ${darken(green, 0.25)};
  }
`;

export default Button;
