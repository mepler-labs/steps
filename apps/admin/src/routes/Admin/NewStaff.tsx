import * as React from 'react';
import Modal from '../../components/Modal';
import styled from 'styled-components';
import ButtonLink from 'atoms/ButtonLink';
import NewStaffForm from './NewStaffForm';

interface Props {
  actions: { createStaff: any };
}

export class NewStaff extends React.Component<Props, {}> {
  // createClient = clientData => {
  //   this.props.actions.createClient(clientData)
  //     .then(res => {
  //       this.setRedirect();
  //       this.renderRedirect();
  //     }).catch(error => this.setState({ shouldRedirect: false, errorMessage: error }))
  // };

  public createStaff({ emails }) {
    const splitEmails = emails.split(/,\s*/);
    this.props.actions.createStaff(splitEmails);
  }

  render() {
    return (
      <StyledModal>
        <div>
          <h1>Invite Staff</h1>
          <p>
            Enter the emails of the people you would like to invite, use commas
            to invite multiple people.
          </p>
          <NewStaffForm onSubmit={this.createStaff} />
        </div>
      </StyledModal>
    );
  }
}

const StyledModal = styled(Modal)`
  div {
    position: relative;
    min-width: 400px;
    max-width: 840px;
    margin: auto;
  }
  margin: auto;
`;

export default NewStaff;
