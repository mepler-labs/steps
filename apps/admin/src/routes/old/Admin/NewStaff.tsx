import { addAlert } from 'actions/alerts';
import { inviteStaff } from 'actions/staff';
import { AlertLevel, Alert } from 'components/Alert/types';
import Modal from 'components/Modal';
import NewStaffForm from 'forms/NewStaffForm';
import { Flex } from 'grid-styled';
import * as React from 'react';
import { connect } from 'react-redux';
import { History } from 'react-router';
import { withRouter } from 'react-router-dom';
import { bindActionCreators } from 'redux';
import styled from 'styled-components';

interface Props {
  actions: { inviteStaff; addAlert: (alert: Alert) => void };
  history: History;
}

export class NewStaff extends React.Component<Props, {}> {
  createStaff = ({ emails }) => {
    const splitEmails = emails.split(/,\s*/);
    this.props.actions
      .inviteStaff(splitEmails)
      .then(_res => {
        this.props.history.push('/admin/staff');
      })
      .catch(err => {
        this.props.actions.addAlert({
          message: err.message,
          level: AlertLevel.Error,
          id: 'new-staff-error',
        });
      });
  };

  render() {
    return (
      <StyledModal>
        <Flex justifyContent="center">
          <NewStaffForm onSubmit={this.createStaff} />
        </Flex>
      </StyledModal>
    );
  }
}

const StyledModal = styled(Modal)`
  div {
    position: relative;
    min-width: 400px;
    max-width: 840px;
    margin: auto;
  }
  margin: auto;
`;

const mapDispatchToProps = dispatch => ({
  actions: bindActionCreators({ inviteStaff, addAlert }, dispatch),
});

export default connect(
  null,
  mapDispatchToProps,
)(withRouter(NewStaff));
