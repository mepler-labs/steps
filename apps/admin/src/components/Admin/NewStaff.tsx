import PageHeader from 'components/Headers/PageHeader';
import Button from 'atoms/Buttons/Button';

interface Props {}
const NewStaff: React.SFC<Props> = props => (
  <div>
    <PageHeader label="Staff">
      <Link to="/admin/staff/new">
        <Button>Invite Staff</Button>
      </Link>
    </PageHeader>
    <Label>Name & Email</Label>
    <StaffList staff={this.props.staff} />
  </div>
);
