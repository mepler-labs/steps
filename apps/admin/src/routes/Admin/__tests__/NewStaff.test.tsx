import { shallow, mount } from 'enzyme';
import { NewStaff } from '../NewStaff';
import * as React from 'react';
import { ReduxForm } from 'redux-form';
import Button from '../../../atoms/button';

describe('NewStaff.tsx', () => {
  it('renders correctly', () => {
    const actions = {
      createStaff: jest.fn()
    };
    const wrapper = shallow(<NewStaff actions={actions} />);

    expect(wrapper).toMatchSnapshot();
  });
});
