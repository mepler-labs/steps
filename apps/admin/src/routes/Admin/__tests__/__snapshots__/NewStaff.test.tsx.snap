// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewStaff.tsx renders correctly 1`] = `
<Styled(Styled(Modal))>
  <div>
    <h1>
      Invite Staff
    </h1>
    <p>
      Enter the emails of the people you would like to invite, use commas to invite multiple people.
    </p>
    <ReduxForm
      onSubmit={[Function]}
    />
  </div>
</Styled(Styled(Modal))>
`;
